### ISSUE TITLE
Improve "run for each" variables logic

### ISSUE THREAD

**Issue by Sh1Yo** [PortSwigger/BChecks#31]
When the `run for each` variables are used the requests are getting repeated even though `run for each` placement may not be reached.

Consider the following situation:

You want to check for xss vulnerabilities. You have 10 payloads in `run for each`. But before sending those payloads you need to make sure whether the injection point reflects on the page to decrease the amount of useless requests

**Comment by A-J-C** [comment]
Yes, currently the run for each will always repeat the whole check.

We would need to introduce a more advanced looping syntax for this use case. I'll keep this issue open to track the need for this feature.
